cli.py

```diff
@@ -0,0 +1,36 @@
+import argparse
+
+from omegaconf import DictConfig, OmegaConf
+from typing import Any, Tuple
+
+
+def load_config(config_path: str) -> DictConfig:
+    """Load a config file of a given path (absolute or relative to cwd)."""
+    conf = OmegaConf.load(config_path)
+    print(f"Loaded config from {config_path}")
+    print(OmegaConf.to_yaml(conf))
+    return conf
+
+
+def parse_arguments() -> Tuple[Any, DictConfig]:
+    """CLI arg parsing"""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--config", type=str, default="configs/mnist.yaml", help="U-method config file"
+    )
+    parser.add_argument(
+        "--dry-run", type=bool, default=False, action="store_true",
+        help="Dry run for debugging. Prints information about the potential run."
+    )
+    parser.add_argument(
+        "--name", type=str, default=None, help="Optionally identify your run"
+    )
+    # Verbose is boolean for now
+    parser.add_argument(
+        "--verbose", action="store_true", help="Increase output verbosity"
+    )
+
+    args = parser.parse_args()
+    config = load_config(args.config)
```

Comment:
Since we'll be trying many things, I'm not sure it makes sense to have many settings stored in config files. Would recommend not using OmegaConfig, instead defining all our arguments here in argparse. Can make them hierarchical, like `--model.path` and `--model.lora.rank`. Then, when loaded into `wandb.init(config=args)`, `wandb.config` will be a hierarchical object of all the settings. That also makes wandb sweeps over these settings nicer to work with.

Reply:
But we define all out model parameters in these config files. Putting these all in through argv sounds super painful

Reply:
If using config files, how do you specify changes for individual experiments (surely, we don't want a config file per experimental condition)? We'd set default values in cli.py for each of the values, so you'd only specify the things you want to be different from default. This makes it especially nice to run sweeps over config values https://docs.wandb.ai/guides/sweeps/walkthrough

Reply:
Hmm..I'm not aware of a way to, through argparse, define an argument such as `"--model"` that adapts to users passing in new tokens like `--model.path=...`. Instead of spelling out every potential param, I have a `json` solution where `"--model"` is a string arg, but users can make the string a json object e.g. `"{'path': '/some/path'}"` and this will be loaded as a dict to conf.

It's worth noting that you can specify wandb sweeps as config files: https://docs.wandb.ai/guides/sweeps/define-sweep-configuration
so I think we should keep config loading as well.

Reply:
Argparse doesn't adapt to new sequences of parameters, but we don't need that functionality. We'd just specify each of the configurable parameters to our experiments exactly once with default values and tooltips in cli.py. Using periods in the parameter values would be purely aesthetic to help separate different clusters of parameters from each other, though we might use some other syntax like underscores.

I'm down to try omegaconf, but it feels messy to have both that kind of a config file and other argparsed args. If you want to use omegaconf files, it might be better to use [hydra](https://hydra.cc/docs/intro/) to automatically merge in the default values from a config file and any updated values from the command line or a sweep. Hydra is kinda officially supported by wandb if you explicitly convert the config to a dict before wandb project initialization (https://docs.wandb.ai/guides/integrations/hydra). Otherwise, you can use OmegaConf.load(), OmegaConf.from_cli(), and OmegaConf.merge() together to implement hydra-like behavior of having default config file values and updated cli values.